target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8071-eap102.dts

```diff
@@ -0,0 +1,687 @@
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:sbl1";
+				reg = <0x0 0x50000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "0:mibib";
+				reg = <0x50000 0x10000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "0:bootconfig";
+				reg = <0x60000 0x20000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "0:bootconfig1";
+				reg = <0x80000 0x20000>;
+				read-only;
+			};
+
+			partition@a0000 {
+				label = "0:qsee";
+				reg = <0xa0000 0x180000>;
+				read-only;
+			};
+
+			partition@220000 {
+				label = "0:qsee_1";
+				reg = <0x220000 0x180000>;
+				read-only;
+			};
+
+			partition@3a0000 {
+				label = "0:devcfg";
+				reg = <0x3a0000 0x10000>;
+				read-only;
+			};
+
+			partition@3b0000 {
+				label = "0:devcfg_1";
+				reg = <0x3b0000 0x10000>;
+				read-only;
+			};
+
+			partition@3c0000 {
+				label = "0:apdp";
+				reg = <0x3c0000 0x10000>;
+				read-only;
+			};
+
+			partition@3d0000 {
+				label = "0:apdp_1";
+				reg = <0x3d0000 0x10000>;
+				read-only;
+			};
+
+			partition@3e0000 {
+				label = "0:rpm";
+				reg = <0x3e0000 0x40000>;
+				read-only;
+			};
+
+			partition@420000 {
+				label = "0:rpm_1";
+				reg = <0x420000 0x40000>;
+				read-only;
+			};
+
+			partition@460000 {
+				label = "0:cdt";
+				reg = <0x460000 0x10000>;
+				read-only;
+			};
+
+			partition@470000 {
+				label = "0:cdt_1";
+				reg = <0x470000 0x10000>;
+				read-only;
+			};
+
+			partition@480000 {
+				label = "0:appsblenv";
+				reg = <0x480000 0x10000>;
+			};
+
+			partition@490000 {
+				label = "0:appsbl";
+				reg = <0x490000 0xc0000>;
+				read-only;
+			};
+
+			partition@550000 {
+				label = "0:appsbl_1";
+				reg = <0x530000 0xc0000>;
+				read-only;
+			};
+
+			partition@610000 {
+				label = "0:art";
+				reg = <0x610000 0x40000>;
+				read-only;
+			};
+
+			partition@650000 {
+				label = "0:ethphyfw";
+				reg = <0x650000 0x80000>;
+				read-only;
+			};
+
+			partition@6d0000 {
+				label = "0:product_info";
+				reg = <0x6d0000 0x80000>;
+				read-only;
+			};
+
+			partition@750000 {
+				label = "priv_data1";
+				reg = <0x750000 0x10000>;
+				read-only;
+			};
+
+			partition@760000 {
+				label = "priv_data2";
+				reg = <0x760000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&blsp1_uart3 {
+	status = "okay";
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&crypto {
+	status = "okay";
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&prng {
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qusb_phy_0 {
+	status = "okay";
+};
+
+&ssphy_0 {
+	status = "okay";
+};
+
+&ssphy_1 {
+	status = "okay";
+};
+
+&usb_0 {
+	status = "okay";
+};
```

Comment:
Ok, so something is not correct here at all.
You state that it has 2x USB2.0 ports but the only enable one USB controller and its HS and SS PHY-s but leave the second one disabled and only enable its SS PHY?

Reply:
Ok, but where does the second USB port come from?
Also, why is SS PHY even enabled if its HS only?

Reply:
Can't answer to either, but without ssphy_0 enabled, both USB ports aren't active.

Reply:
Hm, so they are using some kind of a switch then.
What does lsusb show?

Reply:
With both ports populated with a pl2303 and kingston:
```root@OpenWrt:~# lsusb
Bus 001 Device 005: ID 067b:2303 Prolific Technology Inc.  USB-Serial Controller D
Bus 001 Device 004: ID 2fe3:000b ZEPHYR USB-DEV
Bus 002 Device 001: ID 1d6b:0003 Linux 5.15.19 xhci-hcd xHCI Host Controller
Bus 001 Device 003: ID 0930:6545 Kingston DataTraveler 2.0
Bus 001 Device 002: ID 05e3:0608  USB2.0 Hub
Bus 001 Device 001: ID 1d6b:0002 Linux 5.15.19 xhci-hcd xHCI Host Controller

Reply:
`lsusb -t` to get the tree view please

Reply:
I don't know what to do about `Bus 001 Device 004: ID 2fe3:000b ZEPHYR USB-DEV`. Assuming we aren't supporting bluetooth as on other OpenWrt supported devices?

Reply:
Well yeah, they have a hub, wouldn't make sense otherwise.
The thing with those USB Bluetooth implementations is that they are custom, it's usually a Bluetooth MCU, most likely a TI CC series with custom FW, and it's not supported at all by bluez or anything else so you really have no way of supporting it.